prime-dotnet-webapi/Services/HealthAuthorityService.cs

```diff
@@ -148,17 +154,23 @@ public async Task UpdatePrivacyOfficeAsync(int healthAuthorityId, PrivacyOfficeV
 
         public async Task UpdateVendorsAsync(int healthAuthorityId, IEnumerable<int> vendorCodes)
         {
-            var oldVendors = await _context.HealthAuthorityVendors
+            var existingVendors = await _context.HealthAuthorityVendors
                 .Where(ct => ct.HealthAuthorityOrganizationId == healthAuthorityId)
                 .ToListAsync();
 
-            _context.HealthAuthorityVendors.RemoveRange(oldVendors);
+            var deletedVendors = existingVendors
+                .Except(existingVendors
+                    .Where(ev => vendorCodes.Contains(ev.VendorCode)));
 
-            var newVendors = vendorCodes.Select(code => new HealthAuthorityVendor
-            {
-                HealthAuthorityOrganizationId = healthAuthorityId,
-                VendorCode = code
-            });
+            _context.HealthAuthorityVendors.RemoveRange(deletedVendors);
```

Comment:
You did not guard the backend to prevent deletion of vendor codes/care types that are in use

Reply:
Added guard